Replaced the Button components of Material UI with Button components of @olympusdao/component-library

ashiskumar-1999 opened the pull request:
All the Button components in the src/components have been replaced.

ashiskumar-1999 committed:
Replace all the Material UI button components with button component of "@olympusdao/component-library"

ashiskumar-1999 committed:
Update .eslintrc.js

brightiron commented on src/components/Migration/MigrationModalSingle.tsx:
```diff
@@ -220,17 +209,17 @@ function MigrationModalSingle({ open, handleClose }: { open: boolean; handleClos
                         </Typography>
                       ) : row.fullApproval ? (
-                        <Button
+                        <TextButton
```
Could you explain why this is moving from an outlined button to a text button?

brightiron commented on src/components/GiveProject/ProjectCard.tsx:
```diff
@@ -396,16 +396,15 @@ export default function ProjectCard({ project, mode }: ProjectDetailsProps) {
                 <Grid item xs={6} sm={12} md={6} className="give-button-grid">
-                  <Button
+                  <PrimaryButton
                     variant="contained"
```
please pay attention to props across your PR. variants, and in most cases props are no longer needed when moving from the mui button component to our component-library component.

brightiron commented on src/components/GiveProject/GiveHeader.tsx:
```diff
@@ -27,23 +27,23 @@ export function GiveHeader({ isSmallScreen, isVerySmallScreen, totalDebt, networ
       >
-        <Button color="secondary">
+        <SecondaryButton>
           <Typography variant="h6">
```
most buttons do not need to be wrapped in typography tags unless it significantly alters the style of the button

brightiron reviewed (changes requested):
Please see review feedback. i did not call out everything, but these are persistent themes through your changes. Please review our component-library storybook for docs on proper use of these button components. 
can you also provide screenshots of changes with your PR? Have you verified visually these changes?

ashiskumar-1999 committed:
Remove the varient in the button component and change the templates of the button  components as required.

ashiskumar-1999 committed:
Update .eslintrc.js

ashiskumar-1999 commented:
Hi @brightiron, I have changed the buttons according to the docs and removed the variant props from it. Also, I have removed the typography from the button where it wasn't necessary. You can take a look.

brightiron commented:
@ashiskumar-1999 - thanks. can you please resolve your merge conflicts and once done I will review again

ashiskumar-1999 committed:
Merge branch 'develop' into develop

ashiskumar-1999 committed:
Update MigrationModal.tsx

ashiskumar-1999 committed:
Update MigrationModal.tsx

brightiron commented on src/components/CallToAction/CallToAction.tsx:
```diff
@@ -1,43 +1,41 @@
 
 import { ReactComponent as ArrowUp } from "../../assets/icons/arrow-up.svg";
 
-export const LearnMoreButton = () => (
-  <Button
-    variant="outlined"
-    color="secondary"
-    href="https://docs.olympusdao.finance/main/basics/migration"
-    target="_blank"
-    className="learn-more-button"
-  >
-    <Typography variant="body1">
-      <Trans>Learn More</Trans>
-    </Typography>
-
-    <SvgIcon component={ArrowUp} color="primary" />
-  </Button>
-);
```
ser, have you visually verified your changes? This branch has compiling errors, and once resolved this button looks like this. Can you please provide sceenshots with your changes to visually verify your changes? 
![image](https://user-images.githubusercontent.com/95196612/152709967-bef3cb0b-a18d-487f-a5b1-c29a6fda5fed.png)

ashiskumar-1999 replied:
Hi @brightiron, This button wasn't visible to me because I didn't have any assets. I have changed this according to the material UI properties.

brightiron commented:
![image](https://user-images.githubusercontent.com/95196612/152711040-679b92e1-e151-42f2-bbaf-534b4c865ac8.png)

ashiskumar-1999 committed:
Merge branch 'develop' into develop

ashiskumar-1999 committed:
Merge branch 'develop' into develop
